### ISSUE TITLE
Use an affine map for `Arc` like we do for `Ellipse`.

### ISSUE THREAD

**Issue by derekdreery** [linebender/kurbo#156]
I think representing an ellipse as an affine map of the unit circle has worked well. We could represent an `Arc` like this too.

**Comment by cmyr** [comment]
Indeed, `Arc` feels like a neglected corner of the codebase right now, and I think there's generally room for polish.

**Comment by derekdreery** [comment]
Notes to self: In the way we think of an ellipse as an affine map of the unit circle, we need to think about the Arc as an affine map of part of the unit circle.

**Comment by derekdreery** [comment]
I've actually played with doing this, but there is an issue: let `A` be the affine map associated with the map from the unit circle to an ellipse, and let `USV = A` be a SVD (U, V are orthonormal (ie. rotations), S is singular (i.e. non-uniform scale)). In the `Ellipse` case, you don't care about the initial rotation V, because a rotation to the unit circle is the identity. But you **do** care about it for the `Arc` case, because it changes the start angle. It's not insurmountable, I think you just add the angle associated with V to the start angle, but you need to consider it.

This means that we need the full SVD rather than just the `U` and the `S`, so the SVD method needs to be updated. We should also note that the SVD is not unique, because you can flip the radii and add pi/2 to the x_rotation. It's worth considering storing the start and end angles in Cartesian space (cos and sin of each) rather than the start and sweep angles, because it will mean that calculations don't have to go to polar coordinates, which should improve performance (speed *and* accuracy). 

Finally, applying an affine map to the arc is more complex. I haven't thought about how you would calculate the start/sweep angles, but it is more complicated than just composing two affine maps.

**Comment by derekdreery** [comment]
I think if we do all this we should expose the SVD method - I'm sure decomposing an affine map into rotation then scale then rotation then translation is useful in situations that users may have.
